app/lib/katello/concerns/base_template_scope_extensions.rb

```diff
@@ -45,52 +90,113 @@ def host_products_names(host)
       end
 
+      apipie :method, 'Returns IDs of the applicable errata on the host' do
+        required :host, 'Host::Managed', desc: 'Host object to get the applicable errata for'
+        returns array_of: Integer, desc: 'Array with applicable errata IDs of the host'
+      end
       def host_applicable_errata_ids(host)
         host.applicable_errata.map(&:errata_id)
       end
 
+      apipie :method, 'Returns filtered applicable errata for the host' do
+        required :host, 'Host::Managed', desc: 'Host object to get the applicable errata for'
+        optional :filter, String, desc: 'Filter to apply on applicable errata', default: ''
+        returns array_of: 'Erratum', desc: 'Filtered applicable errata for the host'
+      end
       def host_applicable_errata_filtered(host, filter = '')
         host.applicable_errata.includes(:cves).search_for(filter)
       end
 
+      apipie :method, 'Returns version of the latest applicable RPM package' do
+        required :host, 'Host::Managed', desc: 'Host object to get the applicable RPM package version on'
+        required :package, String, desc: 'Name of the package'
+        returns String, desc: 'Package version'
+      end
       def host_latest_applicable_rpm_version(host, package)
         host.applicable_rpms.where(name: package).order(:version_sortable).limit(1).pluck(:nvra).first
       end
 
+      apipie :method, 'Loads Pool objects' do
+        desc "This macro returns a collection of Pools matching search criteria.
+          The collection's loaded in bulk of 1 000 records."
+        keyword :search, String, desc: 'A search term to limit the resulting collection, using standard search syntax', default: ''
+        keyword :includes, Array, of: [String, Symbol], desc: 'An array of associations represented by strings or symbols, to be included in the SQL query. The list can be extended
+          from plugins and can not be fully documented here. Most used associations are :subscription, :products, :organization', default: nil
+        returns array_of: 'Pool', desc: 'The collection that can be iterated over using each_record'
+      end
       def load_pools(search: '', includes: nil)
         load_resource(klass: Pool.readable, search: search, permission: nil, includes: includes)
       end
 
+      apipie :method, 'Returns the last time the host checked in via RHSM' do
+        required :host, 'Host::Managed', desc: 'Host object to get the last time the host checked in'
+        returns 'ActiveSupport::TimeWithZone', desc: 'The last time the host checked in via RHSM'
+      end
       def last_checkin(host)
         host&.subscription_facet&.last_checkin
       end
 
+      apipie :method, 'Loads Erratum objects' do
+        desc "This macro returns a collection of Errata matching filter criteria.
+          The collection's loaded in bulk of 1 000 records."
+        keyword :filter_errata_type, String, desc: 'TODO: need help', default: nil
```

Comment:
It appears this isn't used anymore.  This should be removed, also line 207 below.

Reply:
I've rebased the PR and the parameter seems to be still present. Should I remove it from the signature and body?

Reply:
Sorry for being unclear! Yes that's what I meant originally.  However, turns out I missed its use on line 247 below.  So we shouldn't remove it.

```suggestion
        keyword :filter_errata_type, String, desc: "Errata type. One of: #{Katello::Erratum::TYPES.join(', ')}", default: nil
```

Reply:
^ Suggested change is in the comment above